### DISCUSSION TITLE
Need to find documentation on the statistic available for vector wind

### DISCUSSION THREAD

**Discussion by Ho-ChunHuang-NOAA** [dtcenter/METplus#1589] [Plot Generation] [answered]
I need to understand what does it mean for statistic options of  vcnt_fbar versu vcnt_fbar_speed, and same for vcbt_obar* for vector wind.  What is the diffference of vl1l2_* and vcnt_* ? Where is the documention for the vector wind statistic graphic?

Example are 20220425_192827 and 20220425_192941.

I tried https://metviewer.readthedocs.io/en/latest/Users_Guide/index.html but no luck.

**Comment by jprestop** [comment]
Hi @Ho-ChunHuang-NOAA.  

Thank you for your question.  I see you'd like to better understand some of the statistics options for winds.  Please read through [Appendix G Vectors and Vector Statistics](https://met.readthedocs.io/en/latest/Users_Guide/appendixG.html?highlight=wind) of the MET User's Guide and see if it answers your questions.  Please follow up with any questions or let us know that you're all set after taking a look.

**Comment by Ho-ChunHuang-NOAA** [comment, marked as answer]
Can you tell me the difference between VCNTT_\* versus VL1L2_*, e.g., VCNT_FBAR and VL1L2FVAR.

Also, what are VL1L2_BIAs and VL1L2_TOTAL?

The document in your reply  seems to be the VCNT_* quantities.  Is there a documentation for VL1L2_*?

**Reply by jprestop** [reply]
Hi @Ho-ChunHuang-NOAA.  Thank you for letting us know that you still have questions.  I need to defer to @JohnHalleyGotway's expertise.  @JohnHalleyGotway, could you please help answer Ho-Chun's questions?

**Reply by JohnHalleyGotway** [reply]
@Ho-ChunHuang-NOAA the "VL1L2_" and "VCNT_" prefix you see in the list of statistics served up via METviewer indicate the line type from which the statistic is retrieved in the MET output files. Here are links to the tables in the MET documentation which define the contents for the vector-based MET output line types: [VL1L2 line type](https://met.readthedocs.io/en/latest/Users_Guide/point-stat.html#id21), [VAL1L2 line type](https://met.readthedocs.io/en/latest/Users_Guide/point-stat.html#id22), and [VCNT line type](https://met.readthedocs.io/en/latest/Users_Guide/point-stat.html#id23).

As @jprestop points out, [Appendix G Vectors and Vector Statistics](https://met.readthedocs.io/en/latest/Users_Guide/appendixG.html?highlight=wind) contains some description of the stats in the VCNT line type. You can find equations for VL1L2 and VAL1L2 partial sums listed in the [Vector L1 and L2 values](https://met.readthedocs.io/en/latest/Users_Guide/appendixC.html#vector-l1-and-l2-values) and [Vector anomaly L1 and L2 values](https://met.readthedocs.io/en/latest/Users_Guide/appendixC.html#vector-anomaly-l1-and-l2-values) sections of the user's guide. Those sections existed prior to adding the VCNT line type... and I can see how it is a bit confusing to have the vector info spread across multiple spots.

To your specific questions...

- What is VL1L2_TOTAL? That is just the contents of the TOTAL column from the VL1L2 line type. So that's column number 25 which [this table](https://met.readthedocs.io/en/latest/Users_Guide/point-stat.html#id21) defines as the:
```Total number of matched pairs of forecast winds (uf, vf) and observation winds (uo, vo)```

- What is VL1L2_BIAS? Now this is a bit confusing. The VL1L2 line type DOES NOT contain a column named bias! So this is a statistic that METviewer is deriving on the fly from the VL1L2 columns that do exist. But what's the equation for that? I had to actually search the METviewer code to find it on [this line](https://github.com/dtcenter/METviewer/blob/fd9f6093c551cac133b77b83e6df9507f506c0bc/R_work/include/statistics.R#L524).

```
calcVL1L2_BIAS = function(d){
  bias = sqrt(d$uvffbar) - sqrt(d$uvoobar);
  return ( round(bias, digits=5)) ;
}
```

Scrolling down in that file you'll find similar derivation functions for other VL1L2 statistics.
Also note that METviewer is transitioning away from R and the statistic derivations have been reimplemented in python. The VL1L2 statistic derivations can be found in the [vl1l2_statistics.py](https://github.com/dtcenter/METcalcpy/blob/main_v1.1/metcalcpy/util/vl1l2_statiatics.py) file in the METcalcpy repo. When you use the python plotting options in METviewer, those are the functions that are called.

To summarize...

- Many statistics are computed by both MET and METviewer (or really METcalcpy now). Equations for all statistics computed by MET should be found in the [MET User's Guide](https://met.readthedocs.io/en/latest/Users_Guide/index.html).

- Some statistics are computed only by METviewer (or really METcalcpy). Those equations *should be* documented in the [METviewer User's Guide](https://metviewer.readthedocs.io/en/latest/Users_Guide/index.html) or [METcalcpy User's Guide](https://metcalcpy.readthedocs.io/en/latest/Users_Guide/index.html)... but clearly they are not. For example, I couldn't find documentation about the derived VL1L2_BIAS.

Since you've been wading through all of these details, do you have any recommendations on how we should organize and document these details?

I lean toward documenting the MET-produced stats in the MET User's Guide and documenting the METcalcpy-derived stats in the METcalcpy User's Guide. But then update the METviewer User's Guide to refer people to the METcalcpy one.

What do you think?

**Reply by JohnHalleyGotway** [reply]
@Ho-ChunHuang-NOAA, since we didn't discuss these details any further at the METplus NOAA telecon yesterday, I assume we can mark this discussion as being answered and lock future conversation on it. But I'll wait until the end of today to do so to give you an opportunity to follow up, if needed.

Thanks!

**Comment by Ho-ChunHuang-NOAA** [comment]
Hi, John:

Thanks.  Please close this ticket.  I was able to identify two variables
(vcnt_rmsve and vl1l2_bias) that I fully understand of what is computed.
For the rest, I will take time to sort them through.

Ho-Chun Huang, Ph.D.
